addons/web/static/src/js/widgets/data_export.js

```diff
@@ -56,10 +56,19 @@ var DataExport = Dialog.extend({
                 }
             }
         },
-        'dblclick .o_export_tree_item': function(e) {
+        'dblclick .o_export_tree_item:not(.haschild)': function(e) {
+            var self = this;
             var $elem = $(e.currentTarget);
             $elem.removeClass('o_selected');
             this.add_field($elem.data('id'), $elem.find('.o_tree_column').first().text());
+            // Add its parents to export
+            var $parents = $elem.parents('.o_export_tree_item');
+            if ($parents.length) {
+                _.each($parents, function (el) {
```

Comment:
Why is exporting a sub-field automatically adding a bunch of irrelevant garbage to the export? In an awful order to boot, and with unclear behaviour: the parent field was made non-selectable, but it's now added and blocks the addition of the semantically identical but nominally separate External ID.

Reply:
If user add child field to export then we automatically add its parent field's `External ID` to export list